plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStorageProcessor.java

```diff
@@ -1849,6 +1858,488 @@ public Answer createSnapshot(final CreateObjectCommand cmd) {
         }
     }
 
+    private SnapshotObjectTO createSnapshotToAndUpdatePathAndSize(String path, String fullPath) {
+        final File snapFile = new File(fullPath);
+        long size = 0;
+
+        if (snapFile.exists()) {
+            size = snapFile.length();
+        }
```

Comment:
if snapshot file doesn't exists, is it Ok to throw error here?

Reply:
If the file does not exist after taking the snapshot, an error should definitely be thrown.